File: mobile/src/main/java/com/backdoor/moove/core/data/ReminderDataProvider.java
package com.backdoor.moove.core.data;

import android.content.Context;
import android.database.Cursor;
import android.support.annotation.Nullable;

import com.backdoor.moove.core.helper.DataBase;

import java.util.ArrayList;

public class ReminderDataProvider {

    public static ArrayList<MarkerModel> getListData(Context context) {
        ArrayList<MarkerModel> list = new ArrayList<>();
        list.clear();
        DataBase db = new DataBase(context);
        db.open();
        Cursor c = db.getAllReminders();
        if (c != null && c.moveToFirst()) {
            do {
                String title = c.getString(c.getColumnIndex(DataBase.SUMMARY));
                long id = c.getLong(c.getColumnIndex(DataBase._ID));
                int icon = c.getInt(c.getColumnIndex(DataBase.MARKER));

                if (icon == -1) {
                    icon = 0;
                }

                list.add(new MarkerModel(title, id, icon));
            } while (c.moveToNext());
        }
        if (c != null) {
            c.close();
        }

        db.close();
        return list;
    }

    public static ArrayList<ReminderModel> load(Context context) {
        ArrayList<ReminderModel> list = new ArrayList<>();
        DataBase db = new DataBase(context);
        db.open();
        Cursor c = db.getAllReminders();
        if (c != null && c.moveToFirst()) {
            do {
                String title = c.getString(c.getColumnIndex(DataBase.SUMMARY));
                String type = c.getString(c.getColumnIndex(DataBase.TYPE));
                String number = c.getString(c.getColumnIndex(DataBase.NUMBER));
                String uuID = c.getString(c.getColumnIndex(DataBase.UUID));
                String melody = c.getString(c.getColumnIndex(DataBase.MELODY));
                long startTime = c.getLong(c.getColumnIndex(DataBase.START_TIME));
                long id = c.getLong(c.getColumnIndex(DataBase._ID));
                int statusDb = c.getInt(c.getColumnIndex(DataBase.STATUS_DB));
                int statusList = c.getInt(c.getColumnIndex(DataBase.STATUS_REMINDER));
                double lat = c.getDouble(c.getColumnIndex(DataBase.LATITUDE));
                double lon = c.getDouble(c.getColumnIndex(DataBase.LONGITUDE));
                int radius = c.getInt(c.getColumnIndex(DataBase.RADIUS));

                list.add(new ReminderModel(title, type, uuID, statusDb, startTime, id,
                        new double[]{lat, lon}, number, statusList, radius, melody));
            } while (c.moveToNext());
        }
        if (c != null) {
            c.close();
        }

        db.close();
        return list;
    }

    @Nullable
    public static ReminderModel getItem(Context mContext, long id) {
        ReminderModel item = null;
        DataBase db = new DataBase(mContext);
        db.open();
        Cursor c = db.getReminder(id);
        if (c != null && c.moveToFirst()) {
            String title = c.getString(c.getColumnIndex(DataBase.SUMMARY));
            String type = c.getString(c.getColumnIndex(DataBase.TYPE));
            String number = c.getString(c.getColumnIndex(DataBase.NUMBER));
            String uuID = c.getString(c.getColumnIndex(DataBase.UUID));
            String melody = c.getString(c.getColumnIndex(DataBase.MELODY));
            long startTime = c.getLong(c.getColumnIndex(DataBase.START_TIME));
            int statusDb = c.getInt(c.getColumnIndex(DataBase.STATUS_DB));
            int statusList = c.getInt(c.getColumnIndex(DataBase.STATUS_REMINDER));
            double lat = c.getDouble(c.getColumnIndex(DataBase.LATITUDE));
            double lon = c.getDouble(c.getColumnIndex(DataBase.LONGITUDE));
            int radius = c.getInt(c.getColumnIndex(DataBase.RADIUS));

            item = new ReminderModel(title, type, uuID, statusDb, startTime, id,
                    new double[]{lat, lon}, number, statusList, radius, melody);
        }
        if (c != null) {
            c.close();
        }

        db.close();
        return item;
    }
}

File: mobile/src/main/java/com/backdoor/moove/core/data/ReminderDataProvider.kt
package com.backdoor.moove.core.data

import android.content.Context
import android.database.Cursor

import com.backdoor.moove.core.helper.DataBase

import java.util.ArrayList

object ReminderDataProvider {

    fun getListData(context: Context): ArrayList<MarkerModel> {
        val list = ArrayList<MarkerModel>()
        list.clear()
        val db = DataBase(context)
        db.open()
        val c = db.allReminders
        if (c != null && c.moveToFirst()) {
            do {
                val title = c.getString(c.getColumnIndex(DataBase.SUMMARY))
                val id = c.getLong(c.getColumnIndex(DataBase._ID))
                var icon = c.getInt(c.getColumnIndex(DataBase.MARKER))

                if (icon == -1) {
                    icon = 0
                }

                list.add(MarkerModel(title, id, icon))
            } while (c.moveToNext())
        }
        c?.close()

        db.close()
        return list
    }

    fun load(context: Context): ArrayList<ReminderModel> {
        val list = ArrayList<ReminderModel>()
        val db = DataBase(context)
        db.open()
        val c = db.allReminders
        if (c != null && c.moveToFirst()) {
            do {
                val title = c.getString(c.getColumnIndex(DataBase.SUMMARY))
                val type = c.getString(c.getColumnIndex(DataBase.TYPE))
                val number = c.getString(c.getColumnIndex(DataBase.NUMBER))
                val uuID = c.getString(c.getColumnIndex(DataBase.UUID))
                val melody = c.getString(c.getColumnIndex(DataBase.MELODY))
                val startTime = c.getLong(c.getColumnIndex(DataBase.START_TIME))
                val id = c.getLong(c.getColumnIndex(DataBase._ID))
                val statusDb = c.getInt(c.getColumnIndex(DataBase.STATUS_DB))
                val statusList = c.getInt(c.getColumnIndex(DataBase.STATUS_REMINDER))
                val lat = c.getDouble(c.getColumnIndex(DataBase.LATITUDE))
                val lon = c.getDouble(c.getColumnIndex(DataBase.LONGITUDE))
                val radius = c.getInt(c.getColumnIndex(DataBase.RADIUS))

                list.add(ReminderModel(title, type, uuID, statusDb, startTime, id,
                        doubleArrayOf(lat, lon), number, statusList, radius, melody))
            } while (c.moveToNext())
        }
        c?.close()

        db.close()
        return list
    }

    fun getItem(mContext: Context, id: Long): ReminderModel? {
        var item: ReminderModel? = null
        val db = DataBase(mContext)
        db.open()
        val c = db.getReminder(id)
        if (c != null && c.moveToFirst()) {
            val title = c.getString(c.getColumnIndex(DataBase.SUMMARY))
            val type = c.getString(c.getColumnIndex(DataBase.TYPE))
            val number = c.getString(c.getColumnIndex(DataBase.NUMBER))
            val uuID = c.getString(c.getColumnIndex(DataBase.UUID))
            val melody = c.getString(c.getColumnIndex(DataBase.MELODY))
            val startTime = c.getLong(c.getColumnIndex(DataBase.START_TIME))
            val statusDb = c.getInt(c.getColumnIndex(DataBase.STATUS_DB))
            val statusList = c.getInt(c.getColumnIndex(DataBase.STATUS_REMINDER))
            val lat = c.getDouble(c.getColumnIndex(DataBase.LATITUDE))
            val lon = c.getDouble(c.getColumnIndex(DataBase.LONGITUDE))
            val radius = c.getInt(c.getColumnIndex(DataBase.RADIUS))

            item = ReminderModel(title, type, uuID, statusDb, startTime, id,
                    doubleArrayOf(lat, lon), number, statusList, radius, melody)
        }
        c?.close()

        db.close()
        return item
    }
}
